workflows/readmapping.nf

```diff
@@ -109,18 +109,19 @@ workflow READMAPPING {
     //
     // ***PacBio condition does not work - needs fixing***
     if ( ch_reads.pacbio || ch_reads.clr ) {
-        if ( params.vector_db.endsWith( '.tar.gz' ) ) {
-            UNTAR ( [ [:], params.vector_db ] ).untar
-            | set { ch_vector_db }
-
+        if ( params.hifi_adapter_db.endsWith( '.tar.gz' ) ) {
+            UNTAR ( [ [:], params.hifi_adapter_db ] ).untar
+            | set { ch_hifi_adapter_db }
             ch_versions = ch_versions.mix ( UNTAR.out.versions )
 
         } else {
-            Channel.fromPath ( params.vector_db )
-            | set { ch_vector_db }
+            Channel.fromPath ( params.hifi_adapter_db )
+            | set { ch_hifi_adapter_db }
         }
     }
 
+    ch_hifi_adapter_yaml = Channel.fromPath ( params.hifi_adapter_yaml )
```

Comment:
```suggestion
    ch_hifi_adapter_yaml = Channel.fromPath ( params.hifi_adapter_yaml ) | collect()
```

or alternatively

```suggestion
    ch_hifi_adapter_yaml = Channel.value ( file(params.hifi_adapter_yaml, checkIfExists: true) )
```

This creates a value channel, which can be re-used multiple times, which will fix your subworkflow input bug